src/components/form/HdInputPassword.vue

```diff
@@ -0,0 +1,236 @@
+<template>
+  <div
+    :class="fieldClasses"
+    class="field field--input"
+  >
+    <img
+      v-if="icon"
+      :src="icon"
+      role="presentation"
+      class="field__icon"
+    >
+    <input
+      :autocomplete="autocomplete"
+      :value="value"
+      :type="currentType"
+      :id="name"
+      :name="name"
+      :placeholder="isActive && placeholder !== undefined ? placeholder : ''"
+      :required="required"
+      :autofocus="autofocus"
+      :disabled="disabled"
+      class="field__input"
+      ref="input"
+      @input="handleInput"
+      @focus="handleFocus"
+      @blur="handleBlur"
+    />
+    <label
+      v-if="label"
+      :for="name"
+      class="field__label"
+    >
+      {{ label }}
+    </label>
+    <p
+      v-if="error"
+      class="field__error"
+    >
+      {{ error }}
+    </p>
+    <p
+      v-else-if="helper"
+      class="field__error field__error--helper"
+      v-html="helper"
+    />
+    <span v-if="showVisibilityToggle"
+      class="field__visibilityToggle"
+      :class="{'field__visibilityToggle--visible': currentType === 'text'}"
+      @click="togglePasswordVisibility"
+    />
+    <span class="field__border"/>
+  </div>
+</template>
+
+<script>
+import merge from 'lodash/merge';
+import { getMessages } from 'hd-blocks/lang';
+
+export default {
+  name: 'HdInputPassword',
+  props: {
+    label: {
+      type: String,
+      default: '',
+    },
+    name: {
+      type: String,
+      required: true,
+    },
+    value: {
+      type: [String, Number],
+      default: '',
+    },
+    placeholder: {
+      type: String,
+      default: '',
+    },
+    required: {
+      type: Boolean,
+      default: false,
+    },
+    autocomplete: {
```

Comment:
Not sure we need all this options?

Reply:
yes, we don't need most of them, I just left them just in case.